pkg/brownfield/pools.go

```diff
@@ -0,0 +1,172 @@
+	var garbage []n.ApplicationGatewayBackendAddressPool
+
+	blacklistedSet := indexPoolsByName(existingBlacklisted)
+	managedSet := indexPoolsByName(managedPools)
+
+	for poolName, pool := range indexPoolsByName(existingNonBlacklisted) {
+		_, existsInBlacklist := blacklistedSet[poolName]
+		_, existsInNewPools := managedSet[poolName]
+		if !existsInBlacklist && !existsInNewPools {
+			garbage = append(garbage, pool)
+		}
+	}
+
+	glog.V(3).Info("[brownfield] Pools AGIC created: ", getPoolNames(managedPools))
+	glog.V(3).Info("[brownfield] Existing Blacklisted Pools AGIC will retain: ", getPoolNames(existingBlacklisted))
+	glog.V(3).Info("[brownfield] Existing Pools AGIC will remove: ", getPoolNames(garbage))
+}
+
+func indexPoolsByName(pools []n.ApplicationGatewayBackendAddressPool) poolsByName {
+	indexed := make(poolsByName)
+	for _, pool := range pools {
+		indexed[backendPoolName(*pool.Name)] = pool
+	}
+	return indexed
+}
+
+func getPoolNames(pool []n.ApplicationGatewayBackendAddressPool) string {
+	var names []string
+	for _, p := range pool {
+		names = append(names, *p.Name)
+	}
+	if len(names) == 0 {
+		return "n/a"
+	}
+	return strings.Join(names, ", ")
+}
+
+func logTarget(verbosity glog.Level, target Target, message string) {
+	t, _ := json.Marshal(target)
+	glog.V(verbosity).Infof("Target is "+message+": %s", t)
+}
+
+// getPoolToTargetsMap creates a map from backend pool to targets this backend pool is responsible for.
+// We rely on the configuration that AGIC has already constructed: Frontend Listener, Routing Rules, etc.
+// We use the Listener to obtain the target hostname, the RoutingRule to get the
+func (c PoolContext) getPoolToTargetsMap() poolToTargets {
+
+	// Index listeners by their name
+	listenersByName := make(map[listenerName]n.ApplicationGatewayHTTPListener)
+	for _, listener := range c.Listeners {
+		listenersByName[listenerName(*listener.Name)] = listener
+	}
+
+	// Index URLPathMaps by the path map name
+	pathNameToPath := make(map[pathmapName]n.ApplicationGatewayURLPathMap)
+	for _, pm := range c.PathMaps {
+		pathNameToPath[pathmapName(*pm.Name)] = pm
+	}
+
+	poolToTarget := make(poolToTargets)
+
+	for _, rule := range c.RoutingRules {
+		listenerName := listenerName(utils.GetLastChunkOfSlashed(*rule.HTTPListener.ID))
+
+		var hostName string
+		if listener, found := listenersByName[listenerName]; !found {
+			continue
+		} else {
+			hostName = *listener.HostName
+		}
+
+		target := Target{Hostname: hostName}
+		if rule.URLPathMap == nil {
+			// SSL Redirects do not have BackendAddressPool
+			if rule.BackendAddressPool == nil {
+				continue
+			}
+			poolName := backendPoolName(utils.GetLastChunkOfSlashed(*rule.BackendAddressPool.ID))
+			poolToTarget[poolName] = append(poolToTarget[poolName], target)
+		} else {
+			// Follow the path map
+			pathMapName := pathmapName(utils.GetLastChunkOfSlashed(*rule.URLPathMap.ID))
```

Comment:
I think we are missing `pathNameToPath[pathMapName].DefaultBackendPool`

Reply:
Agreed - `DefaultBackendAddressPool` needs to be accounted for here